applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt.c

```diff
@@ -200,8 +210,126 @@ void bt_mgmt_conn_disconnect(struct bt_conn *conn, uint8_t reason)
 	}
 }
 
+static void bt_enabled_cb(int err)
+{
+	if (err) {
+		LOG_ERR("Bluetooth init failed (err code: %d)", err);
+		ERR_CHK(err);
+	}
+	k_sem_give(&sem_bt_enabled);
+
+	LOG_DBG("Bluetooth initialized");
+}
+
+static int bonding_clear_check(void)
+{
+	int ret;
+	bool pressed;
+
+	ret = button_pressed(BUTTON_5, &pressed);
+	if (ret) {
+		return ret;
+	}
+
+	if (pressed) {
+		if (IS_ENABLED(CONFIG_SETTINGS)) {
+			LOG_INF("Clearing all bonds");
+			bt_unpair(BT_ID_DEFAULT, NULL);
+		}
+	}
+	return 0;
+}
+
+static void mac_print(void)
+{
+	int ret;
+	/* It may be confusing to print the MAC when privacy is enabled.
+	 * This address will change. Have a look at BT_LOG_SNIFFER_INFO for debugging
+	 * and sniffer purposes
+	 */
+
+	char dev[BT_ADDR_LE_STR_LEN];
+	/* This call will create a new RPA */
+	ret = bt_le_oob_get_local(BT_ID_DEFAULT, &_oob);
+	ERR_CHK(ret);
+
+	(void)bt_addr_le_to_str(&_oob.addr, dev, BT_ADDR_LE_STR_LEN);
+	if (IS_ENABLED(CONFIG_BT_PRIVACY)) {
+		LOG_INF("MAC: %s Valid until next RPA timeout", dev);
+	} else {
+		LOG_INF("MAC: %s", dev);
+	}
+}
+
+static int random_static_addr_cfg(void)
+{
+	int ret;
+	static bt_addr_le_t addr;
+
+	if ((NRF_FICR->INFO.DEVICEID[0] != UINT32_MAX) ||
+	    ((NRF_FICR->INFO.DEVICEID[1] & UINT16_MAX) != UINT16_MAX)) {
+		/* Put the device ID from FICR into address */
+		sys_put_le32(NRF_FICR->INFO.DEVICEID[0], &addr.a.val[0]);
+		sys_put_le16(NRF_FICR->INFO.DEVICEID[1], &addr.a.val[4]);
+
+		/* The FICR value is a just a random number, with no knowledge
+		 * of the Bluetooth Specification requirements for random
+		 * static addresses.
+		 */
+		BT_ADDR_SET_STATIC(&addr.a);
+
+		addr.type = BT_ADDR_LE_RANDOM;
+
+		ret = bt_id_create(&addr, NULL);
```

Comment:
> Generating random static address or random IRK is not supported when calling
 this function before bt_enable().

From the documentation of bt_id_create. This is now called before bt_enable(). Was this not true?

Reply:
Discuss

Reply:
Leave be for this PR.